src/components/fields/image-upload/image-manager/image-manager.ts

```diff
@@ -110,6 +110,7 @@ export const ImageManager = (props: IProps) => {
 								});
 								if (image.addedFrom !== "schema") {
 									compress ? compressImage(index, image) : convertImage(index, image);
+									extractImageMetadata(index, image);
```

Comment:
can find a way to optimise this to avoid setting state twice for the same process?
now it's once in compress/convert image
once in extract data

Reply:
that way you can also extract before compressing